pkg/discovery/discovery.go

```diff
@@ -227,38 +225,47 @@ func (a *apiDiscovery) processSoap(ext *ServiceDetail, svc service.Service) (*Se
 	url := a.cfg.Host + endpoint
 	ep := util2.CreateEndpoint(url)
 	ext.Endpoints = append(ext.Endpoints, ep)
-	return ext, nil
+	return nil
 }
 
-func (a *apiDiscovery) processOAS(ext *ServiceDetail, svc service.Service, p *policy.PolicyItem) (*ServiceDetail, error) {
+func (a *apiDiscovery) processService(ext *ServiceDetail, svc service.Service, p *policy.PolicyItem) error {
+	endpoint := a.getAPIEndpoint(svc.ServiceDetail.ServiceMappings)
+	if endpoint == "" {
+		return fmt.Errorf("unable to find proxy endpoint for %s", svc.ServiceDetail.Name)
+	}
+	ext.Endpoint = endpoint
+	url := a.cfg.Host + endpoint
+	ep := util2.CreateEndpoint(url)
+	ext.Endpoints = append(ext.Endpoints, ep)
+
 	resourceType, docHost := a.getOASEndpoint(p.Policy.All.SetVariable)
 
-	if resourceType == "" {
-		return ext, fmt.Errorf("unable to determine the resource type for %s", svc.ServiceDetail.Name)
+	if resourceType != "" && docHost != "" {
+		// OAS Spec
+		return a.processOAS(ext, resourceType, docHost)
 	}
 
-	if docHost == "" {
-		return ext, fmt.Errorf("api spec not found for %s", svc.ServiceDetail.Name)
+	// Unstructured
+	unstructuredSvc, err := xml.MarshalIndent(svc, "", "    ")
```

Comment:
Do we need to use the Indent call, would that not be adding extra spaces to the byte array?

Reply:
It would add extra spaces in buffer, however without the indentation the UI does not really render formatted xml for unstructured types. Will reduce the indent to 2 spaces

Reply:
Ahh, it's the UI rendering, that makes sense.